resourcescaler.go

```diff
@@ -115,117 +119,121 @@ func (s *AppResourceScaler) GetConfig() (*scaler_types.ResourceScalerConfig, err
+			"path":  lastScaleStatePath,
+			"value": string(scaler_types.ScaleFromZeroStartedScaleEvent),
+		})
+		jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
+			"op":    "add",
+			"path":  lastScaleStateTimePath,
+			"value": string(marshaledTime),
+		})
+	}
+
 	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
 		"op":    "add",
 		"path":  "/status/state",
 		"value": "waitingForProvisioning",
 	})
-	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
-		"op":    "add",
-		"path":  scaleToZeroStatusPath,
-		"value": map[string]interface{}{},
-	})
-	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
-		"op":    "add",
-		"path":  lastScaleStatePath,
-		"value": string(scaler_types.ScaleFromZeroStartedScaleEvent),
-	})
-	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
-		"op":    "add",
-		"path":  lastScaleStateTimePath,
-		"value": string(marshaledTime),
-	})
-
 	err = s.patchIguazioTenantAppServiceSets(namespace, jsonPatchMapper)
 
 	if err != nil {
 		return errors.Wrap(err, "Failed to patch iguazio tenant app service sets")
 	}
 
-	err = s.waitForServiceState(namespace, serviceName, "ready")
+	err = s.waitForServicesState(namespace, serviceNames, "ready")
 
 	if err != nil {
-		return errors.Wrap(err, "Failed to wait for service readiness")
+		return errors.Wrap(err, "Failed to wait for services readiness")
 	}
 
 	return nil
 }
 
-func (s *AppResourceScaler) scaleServiceToZero(namespace string, serviceName string) error {
+func (s *AppResourceScaler) scaleServicesToZero(namespace string, serviceNames []string) error {
 	var jsonPatchMapper []map[string]interface{}
-	s.logger.DebugWith("Scaling to zero", "namespace", namespace, "serviceName", serviceName)
-	desiredStatePath := fmt.Sprintf("/spec/spec/tenants/0/spec/services/%s/desired_state", string(serviceName))
-	markForRestartPath := fmt.Sprintf("/spec/spec/tenants/0/spec/services/%s/mark_for_restart", string(serviceName))
-	scaleToZeroStatusPath := fmt.Sprintf("/status/services/%s/scale_to_zero", string(serviceName))
-	lastScaleStatePath := fmt.Sprintf("/status/services/%s/scale_to_zero/last_scale_event", string(serviceName))
-	lastScaleStateTimePath := fmt.Sprintf("/status/services/%s/scale_to_zero/last_scale_event_time", string(serviceName))
+	s.logger.DebugWith("Scaling to zero", "namespace", namespace, "serviceNames", serviceNames)
 	marshaledTime, err := time.Now().MarshalText()
 	if err != nil {
 		return errors.Wrap(err, "Failed to marshal time")
 	}
-	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
-		"op":    "add",
-		"path":  desiredStatePath,
-		"value": "scaledToZero",
-	})
-	jsonPatchMapper = append(jsonPatchMapper, map[string]interface{}{
-		"op":    "add",
-		"path":  markForRestartPath,
-		"value": false,
-	})
+	for _, serviceName := range serviceNames {
+		desiredStatePath := fmt.Sprintf("/spec/spec/tenants/0/spec/services/%s/desired_state", string(serviceName))
```

Comment:
maybe worth moving those repeated strings outside so both scale from/to would be able to reuse it